addons/udes_security/tests/test_allowed_file_type.py

```diff
@@ -0,0 +1,148 @@
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestAllowedFileType(common.SavepointCase):
+    """
+    Check that only the admin user can manage allowed file types and that downloads are blocked
+    for non-admin users when trying to download files with a file type that is not allowed
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create allowed file type and attachments for blocked/unblocked files
+        """
+        super(TestAllowedFileType, cls).setUpClass()
+
+        cls.AllowedFileType = cls.env["udes.allowed_file_type"]
+        
+        cls.allowed_file_type_csv = cls.env.ref("udes_security.udes_allowed_file_type_csv")
+        cls.allowed_file_type_txt = cls.env.ref("udes_security.udes_allowed_file_type_txt")
+        cls.allowed_file_type_json = cls.env.ref("udes_security.udes_allowed_file_type_json")
+
+        # Set TXT and JSON allowed file types to inactive, so they are blocked for the tests
+        cls.allowed_file_type_txt.active = False
+        cls.allowed_file_type_json.active = False
+
+        cls.atch_txt_file = cls.create_attachment("text_file.txt")
+        cls.atch_csv_file = cls.create_attachment("spreadsheet_file.csv")
+
+    @classmethod
+    def _get_file_path(self, filename):
+        """Get file path from supplied filename"""
+        module_file = sys.modules[self.__module__].__file__
+        module = get_resource_from_path(module_file)[0]
+        files_path_str = get_resource_path(module, "tests", "files")
+        files_path = pathlib.Path(files_path_str)
+        file_path = files_path.joinpath(filename)
+
+        return file_path
+    
+    @classmethod
+    def create_attachment(self, filename):
+        """Create attachment from supplied filename"""
+        IrAttachment = self.env["ir.attachment"]
+
+        file_path = self._get_file_path(filename)
+
+        attachment = IrAttachment.create(
+            {
+                "name": file_path.name,
+                "datas_fname": file_path.name,
+                "datas": base64.b64encode(file_path.read_bytes()),
+            }
+        )
+        return attachment
+
+    @classmethod
+    def update_attachment(self, attachment, filename):
+        """Update attachment data from supplied filename"""
+        file_path = self._get_file_path(filename)
+
+        res = attachment.write({
+            "name": file_path.name,
+            "datas_fname": file_path.name,
+            "datas": base64.b64encode(file_path.read_bytes()),
+        })
+
+        return res
+
+    @classmethod
+    def create_allowed_file_type(self, file_type, **kwargs):
+        """Create allowed file type record from supplied file type"""
+        allowed_file_type_vals = {"name": file_type}
+        allowed_file_type_vals.update(kwargs)
+
+        allowed_file_type = self.AllowedFileType.create(allowed_file_type_vals)
+
+        return allowed_file_type
+
+    def test_allowed_file_active_status(self):
```

Comment:
Nit: the comments are perhaps redundant?  a descriptive method name and docstring ought to be sufficient to document what is being tested.

Reply:
Have renamed this test, although still not convinced about it. Have removed the comments also and added a more descriptive docstring instead.

I've also fixed an issue where it was checking if the TXT file was active rather than the CSV file, and also added in a check for RTF files (which are allowed), to show that both active/inactive attachments are unaffected if a different allowed file type record is made inactive.

Reply:
Maybe split the test into two methods, as it's testing two things?  That should make it easier to name.

Reply:
I've now split the test into `test_assert_allowed_attachment_set_active` and `test_assert_blocked_attachment_set_inactive`. 
I also removed `test_allowed_file_type_deleted` as this is now covered in `test_assert_blocked_attachment_set_inactive`.